src/server/engine_shard_set.cc

```diff
@@ -496,50 +495,34 @@ void EngineShard::PollExecution(const char* context, Transaction* trans) {
 
   // Progress on the transaction queue if no transaction is running currently.
   Transaction* head = nullptr;
-  if (continuation_trans_ == nullptr) {
-    while (!txq_.Empty()) {
-      // Break if there are any awakened transactions, as we must give way to them
-      // before continuing to handle regular transactions from the queue.
-      if (blocking_controller_ && blocking_controller_->HasAwakedTransaction())
-        break;
-
-      head = get<Transaction*>(txq_.Front());
-
-      bool is_armed = head->IsArmedInShard(sid);
-      VLOG(2) << "Considering head " << head->DebugId() << " isarmed: " << is_armed;
-
-      // If the transaction isn't armed yet, it will be handled by a successive poll
-      if (!is_armed)
-        break;
+  while (continuation_trans_ == nullptr && !txq_.Empty()) {
+    // Break if there are any awakened transactions, as we must give way to them
+    // before continuing to handle regular transactions from the queue.
+    if (blocking_controller_ && blocking_controller_->HasAwakedTransaction())
+      break;
 
-      // Avoid processing the caller transaction below if we found it in the queue,
-      // because it most likely won't have enough time to arm itself again.
-      if (head == trans)
-        trans = nullptr;
+    head = get<Transaction*>(txq_.Front());
 
-      TxId txid = head->txid();
+    VLOG(2) << "Considering head " << head->DebugId() << " isarmed: " << head->IsArmedInShard(sid);
 
-      // Update commited_txid before running, because RunInShard might block on i/o.
-      // This way scheduling transactions won't see an understated value.
-      DCHECK_LT(committed_txid_, txid);  //  strictly increasing when processed via txq
-      committed_txid_ = txid;
+    // If the transaction isn't armed yet, it will be handled by a successive poll
+    if (!head->IsArmedInShard(sid))
+      break;
 
-      if (VLOG_IS_ON(2)) {
```

Comment:
This is a small diff in the refactor - that the `DLOG_IF` below becomes de-facto log-level 1. I don't think it's crucial, but you can pass log level as an argument to `run()` if you think it is.

Reply:
They are inconsistent, above and below it uses level 1. I chose the most common one (ooo transactions are very common)